colossalai/shardformer/layer/attn.py

```diff
@@ -197,6 +206,15 @@ def prepare_attn_kwargs(
         if invert:
             attention_mask = invert_mask(attention_mask).unsqueeze(1)
         outputs["attention_mask"] = attention_mask
+
+        element_size = torch.tensor([], dtype=dtype).element_size()
+        memory_size = s_q * s_kv * element_size
+        if memory_size > MEMORY_BOUND:
+            attention_mask = torch.empty((0,), dtype=dtype, device=device)
+            outputs["attention_mask"] = attention_mask
+            if outputs["attention_mask_type"] != AttnMaskType.PADDED_CAUSAL:
+                outputs["attention_mask_type"] = AttnMaskType.CAUSAL
```

Comment:
If memory exceeds bound and the mask type is padded/custom, then let it oom. Becuase dao's flash-attn does not support these two modes.